test/mainnet-fork/Uniswapv4.t.sol

```diff
@@ -0,0 +1,2018 @@
+        mainnetController.mintPositionUniswapV4({
+            poolId     : _POOL_ID,
+            tickLower  : 0,
+            tickUpper  : 11,
+            liquidity  : 0,
+            amount0Max : 0,
+            amount1Max : 0
+        });
+    }
+
+    function test_mintPositionUniswapV4_revertsWhenTickRangeTooWide() external {
+        vm.prank(SPARK_PROXY);
+        mainnetController.setUniswapV4TickLimits(_POOL_ID, -10, 10, 10);
+
+        vm.prank(relayer);
+        vm.expectRevert("MC/tickWidth-too-wide");
+
+        mainnetController.mintPositionUniswapV4({
+            poolId     : _POOL_ID,
+            tickLower  : -8,
+            tickUpper  : 8,
+            liquidity  : 0,
+            amount0Max : 0,
+            amount1Max : 0
+        });
+    }
+
+    function test_mintPositionUniswapV4_revertsWhenAmount0MaxSurpassed() external {
+        vm.prank(SPARK_PROXY);
+        mainnetController.setUniswapV4TickLimits(_POOL_ID, -60, 60, 20);
+
+        ( uint256 amount0Forecasted, uint256 amount1Forecasted ) = _quoteLiquidity(
+            -10,
+            0,
+            1_000_000e6
+        );
+
+        deal(address(usdc), address(almProxy), usdc.balanceOf(address(almProxy)) + amount0Forecasted + 1);
+        deal(address(usdt), address(almProxy), usdt.balanceOf(address(almProxy)) + amount1Forecasted + 1);
+
+        vm.prank(relayer);
+        vm.expectRevert(); // Insufficient allowance.
+        mainnetController.mintPositionUniswapV4({
+            poolId     : _POOL_ID,
+            tickLower  : -10,
+            tickUpper  : 0,
+            liquidity  : 1_000_000e6,
+            amount0Max : amount0Forecasted - 1,
+            amount1Max : amount1Forecasted
+        });
+    }
+
+    function test_mintPositionUniswapV4_revertsWhenAmount1MaxSurpassed() external {
+        vm.prank(SPARK_PROXY);
+        mainnetController.setUniswapV4TickLimits(_POOL_ID, -60, 60, 20);
+
+        ( uint256 amount0Forecasted, uint256 amount1Forecasted ) = _quoteLiquidity(
+            -10,
+            0,
+            1_000_000e6
+        );
+
+        deal(address(usdc), address(almProxy), usdc.balanceOf(address(almProxy)) + amount0Forecasted + 1);
+        deal(address(usdt), address(almProxy), usdt.balanceOf(address(almProxy)) + amount1Forecasted + 1);
+
+        vm.prank(relayer);
+        vm.expectRevert(); // Insufficient allowance.
+        mainnetController.mintPositionUniswapV4({
+            poolId     : _POOL_ID,
+            tickLower  : -10,
+            tickUpper  : 0,
+            liquidity  : 1_000_000e6,
+            amount0Max : amount0Forecasted,
+            amount1Max : amount1Forecasted - 1
+        });
+    }
+
+    function test_mintPositionUniswapV4_revertsWhenRateLimitExceeded() external {
+        vm.startPrank(SPARK_PROXY);
+        mainnetController.setUniswapV4TickLimits(_POOL_ID, -60, 60, 20);
+        rateLimits.setRateLimitData(_DEPOSIT_LIMIT_KEY, 499.966110e18, uint256(499.966110e18) / 1 days);
```

Comment:
Align

Reply:
Should also assert with `_getAmountsForLiquidity` to check that the underlying asset amounts match what was taken